contracts/finance/VestingWalletConfidentialFactory.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {FHE, euint64, externalEuint64, euint128, ebool} from "@fhevm/solidity/lib/FHE.sol";
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {IConfidentialFungibleToken} from "../interfaces/IConfidentialFungibleToken.sol";
+import {VestingWalletCliffConfidential} from "./VestingWalletCliffConfidential.sol";
+import {VestingWalletConfidential} from "./VestingWalletConfidential.sol";
+import {VestingWalletExecutorConfidential} from "./VestingWalletExecutorConfidential.sol";
+
+abstract contract VestingWalletConfidentialFactory {
+    address private immutable _vestingWalletConfidentialImplementation;
+
+    error VestingWalletConfidentialInvalidDuration();
+    error VestingWalletConfidentialInvalidStartTimestamp(address beneficiary, uint64 startTimestamp);
+
+    /**
+     * @dev
+     */
+    event VestingWalletConfidentialBatchFunded(ebool success);
+    /**
+     * @dev
+     */
+    event VestingWalletConfidentialCreated(
+        address indexed beneficiary,
+        address indexed vestingWalletConfidential,
+        uint48 startTimestamp
+    );
+
+    /**
+     * @dev
+     */
+    struct VestingPlan {
+        address beneficiary;
+        externalEuint64 encryptedAmount;
+        uint48 startTimestamp;
+        uint48 cliff;
+        address executor;
+    }
+
+    /**
+     * @dev
+     */
+    constructor() {
+        _vestingWalletConfidentialImplementation = address(new VestingWalletCliffExecutorConfidential());
+    }
+
+    /**
+     * @dev Batches the funding of multiple confidential vesting wallets.
+     *
+     * Funds are sent to predeterministic wallet addresses. Wallets can be created later.
+     */
+    function batchFundVestingWalletConfidential(
+        address confidentialFungibleToken,
+        externalEuint64 totalEncryptedAmount,
+        bytes calldata inputProof,
+        VestingPlan[] calldata vestingPlans,
```

Comment:
We should emit the details of each vesting plan to make it easier for integrators to prompt users to create the wallet.